amida-tech/indaba-client
# Feature/inba 971 fix filtered input / remove grommet search for user groups

#### What does this PR do?
1) Fixes the Filtered Input (particularly for the `SelectUserGroups` modal).
2) Slightly improves Search.
3) Changes placeholders, function names and action types to say "filter" where it's a filter (instead of "search"). 
4) Removes the filter for User Groups for now.

The distinction being that a filter merely parses a distinct set of data, where as search checks the entire set. 


#### Related JIRA tickets:
INBA-971
INBA-978 (Riley reported an error if you select users and then delete the group, the bug has been logged)
INBA-977 (React-Select has a few hiccups. I've touched up one area, but will review this in depth during this ticket)

#### How should this be manually tested?
Check the following filters:
Add New User Group Modal (Wizard and Project Management, both sides)
All Subjects (Subjects Tab)
Projects Filter (Project Manager Dashboard/Home Screen)
Tasks Filter (User Dashboard/Home Screen)
User Sidebar (Project Management, Workflow tab, on the right)
Subjects (Project Management and Wizard)

And check that the filters for User Groups are gone.

Finally, just do a quick regression test of the Search component in:
Add User (Wizard and Project Management) 
ToField (Messaging)
All Users (Users Tab)
Stage Slot (Click on "Assign Task" and a search bar will appear, allowing you to choose a user)
#### Background/Context

#### Screenshots (if appropriate):
